api/src/main/kotlin/bomber/security/TokenAuthenticationFilter.kt

```diff
@@ -0,0 +1,51 @@
+package bomber.security
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
+import org.springframework.util.StringUtils
+import org.springframework.web.filter.OncePerRequestFilter
+import java.io.IOException
+import javax.servlet.FilterChain
+import javax.servlet.ServletException
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+
+class TokenAuthenticationFilter: OncePerRequestFilter() {
+
+    @Autowired
+    private val tokenProvider: TokenProvider? = null
+
+    @Autowired
+    private val customUserDetailsService: CustomUserDetailsService? = null
+
+    @Throws(ServletException::class, IOException::class)
+    override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain) {
+        try {
+            val jwt = getJwtFromRequest(request)
+
+            if (StringUtils.hasText(jwt) && tokenProvider!!.validateToken(jwt!!)) {
+                val userId = tokenProvider.getUserIdFromToken(jwt)
+
+                val userDetails = customUserDetailsService!!.loadUserById(userId)
+                val authentication = UsernamePasswordAuthenticationToken(userDetails, null, userDetails.getAuthorities())
+                authentication.details = WebAuthenticationDetailsSource().buildDetails(request)
+
+                SecurityContextHolder.getContext().authentication = authentication
+            }
+        } catch (ex: Exception) {
+            logger.error("Could not set user authentication in security context", ex)
+        }
+
+        filterChain.doFilter(request, response)
+    }
+
+    private fun getJwtFromRequest(request: HttpServletRequest): String? {
+        val bearerToken = request.getHeader("Authorization")
+        return if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```

Comment:
what will happen if the token looks like this:
```
Bearer  
```

Reply:
after key word Bearer 2 space syms

Reply:
true

Reply:
I fix it with

Reply:
I can made a substring after last space after bearer.

Reply:
Fixed it